test/API/H5_api_test_util.c

```diff
@@ -797,3 +935,223 @@ remove_test_file(const char *prefix, const char *filename)
 
     return ret_value;
 }
+
+int H5_api_test_display_information(void) {
```

Comment:
Fine to move all of the parts of the main function in H5_api_test.c into helper functions, but they should all still be in that file and not moved out to a utility routines file. Unrelated functionality like the seeding of the random number generator below also shouldn't be part of a function called `H5_api_test_display_information`, since it's hiding side effects.